src/dmd/dsymbolsem.d

```diff
@@ -1148,9 +1148,17 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 bool flag;
                 AliasDeclaration ad = imp.aliasdecls[i];
                 //printf("\tImport %s alias %s = %s, scope = %p\n", toPrettyChars(), aliases[i].toChars(), names[i].toChars(), ad._scope);
-                if (imp.mod.search(imp.loc, imp.names[i]) /*, IgnorePrivateImports*/)
+                Dsymbol importedSymbol = imp.mod.search(imp.loc, imp.names[i] /*, IgnorePrivateImports */);
```

Comment:
The comment was added with https://github.com/dlang/dmd/pull/7760.

We should at some point change the logic from pulling out a symbol and then checking it's visibility, to passing the visibility level to search and reduce it with according to import protection (like a funnel).
This would also be necessary to fix `package import pkg.common;`, but requires quite some changes and partial order of visibility.